src/app/public/src/modules/shared/stache-search-results-provider.ts

```diff
@@ -0,0 +1,130 @@
+import { Injectable } from '@angular/core';
+import { SkyAuthHttp } from '@blackbaud/skyux-builder/runtime';
+import { StacheConfigService } from './config.service';
+import 'rxjs/add/operator/toPromise';
+
+@Injectable()
+export class StacheSearchResultsProvider {
+
+  private static readonly STACHE_QUERY_URL:
+  string = 'https://stache-search-query.sky.blackbaud.com';
+  private static readonly STACHE_QUERY_ENDPOINTS:
+  any = {
+    internal: 'query',
+    public: 'query-public'
+  };
+  private isInternal: boolean = true;
+  private siteNames: string[] = [];
+  private endpoints: any;
+  private url: string;
+
+  constructor(
+    private http: SkyAuthHttp,
+    private config: StacheConfigService
+  ) {
+    this.initIsInternal(this.config);
+    this.initEndpoints(this.config);
+    this.initSiteNames(this.config);
+    this.initUrl(this.config);
+  }
+
+  public getSearchResults(searchArgs: any) {
+    if (!this.checkConfig(this.config)) {
+      console.error('Must provide valid configuration for search.');
+      return Promise.reject('Need config in search results provider');
+    }
+
+    let body = {
+      site_names: this.siteNames,
+      search_string: searchArgs.searchText
+    };
+
+    return this.http.post(
+      this.getQueryUrl(),
+      body
+    ).toPromise()
+      .then(res => {
+        let response = res.json();
+        if (response.count === 0) {
+          return {
+            items: [{ title: 'No results found.'}]
+          };
+        }
+        let results = {
+          items: response.results.map((item: any) => {
+            return {
+              title: item.document.title,
+              subtitle: item.document.site_name,
+              description: this.parseDescription(item),
+              url: item.document.host + '/' + item.document.site_name + item.document.path
+            };
```

Comment:
It would benefit consuming classes/apps to provide an interface for the results.

Reply:
I'm the consuming class/app in this case. I'm conforming to what the Omnibar search provider is expecting.

Reply:
I'm sure an interface will get added for that at some point.